integration-app/src/main/java/uk/gov/crowncommercial/esourcing/auth/IpAddressFilter.java

```diff
@@ -0,0 +1,97 @@
+package uk.gov.crowncommercial.esourcing.auth;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Set;
+import javax.annotation.PostConstruct;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.filter.GenericFilterBean;
+
+/**
+ * Standard HTTP Servlet Filter that checks the HTTP requests source IP address is in a specified
+ * list for simple allow list gating.
+ */
+public class IpAddressFilter extends GenericFilterBean {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(IpAddressFilter.class);
+
+  @Value("${ccs.esourcing.tenders.ipallowlist:}")
+  private Set<String> ipAllowList;
+
+  public IpAddressFilter() {}
+
+  @PostConstruct
+  public void init() {
+    if (ipAllowList.isEmpty()) {
+      LOGGER.warn(
+          "No Allow List IP addresses have been specified so requests from all IP addresses will be accepted");
+    } else {
+      LOGGER.info("IP Allow List configuration: {}", ipAllowList);
+    }
+  }
+
+  @Override
+  public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+      throws IOException, ServletException {
+    doFilter((HttpServletRequest) request, (HttpServletResponse) response, chain);
+  }
+
+  private void doFilter(HttpServletRequest request, HttpServletResponse response, FilterChain chain)
+      throws IOException, ServletException {
+
+    // TODO (pillingworth, 2020-01-08) make the paths checked by this filter configurable or use
+    // proper servlet filters and map to a path properly - probably means not adding in as a auth
+    // filter
+    if (request.getServletPath().startsWith("/actuator")) {
+
+      LOGGER.debug("Allowing request on path {}", request.getServletPath());
+
+    } else {
+
+      String remoteAddr = request.getRemoteAddr();
+      String addressToValidate = remoteAddr;
```

Comment:
Is remoteAddr needed?

Reply:
It validates the first address in the X-Forwarded-For and if not present then the remoteAddr against  what is in the allow list. I will merge in for now and will come back and revisit and probably swap the logic around and add some comments to make clearer. There also a TODO in there that needs addressing and I need to add some testing.